_rules/active-focus-shortcut-1e9941.md

```diff
@@ -0,0 +1,255 @@
+---
+id: 1e9941
+name: Printable key shortcut is only active when component has focus
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts for a user interface component are implemented using only printable characters, then they are only available when that component has focus.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with at least one [keyboard shortcut][] that requires pressing only [printable character][] keys to trigger an action on a [user interface component][].
+
+## Expectation
+
+For each [user interface component][] that is a [descendant][] of the root node of the test target, the [printable character][] [shortcut][keyboard shortcut] that triggers the action of that [user interface component][] can only be triggered when that component has [focus][].

```

Comment:
The formulation is not correct, we need to go back to the "if a character key shortcut exist" formulation of before. Or maybe even to another quantification (yiks!) "each character key shortcut…"

Reason: this is looking at every single UI component. Some may have only one character key shortcut and thus "**the** printable character shortcut" is well defined. But some might hav+e either zero, or several of these and then "**the**" is not correctly defined and cannot be used. And to pass the rule, I guess we want **every** printable character shortcut to only trigger when focused.